core/pkg/eval/json_evaluator.go

```diff
@@ -306,6 +310,15 @@ func (je *JSONEvaluator) evaluateVariant(reqID string, flagKey string, context m
 			return "", flag.Variants, model.ErrorReason, metadata, errors.New(model.ParseErrorCode)
 		}
 
+		if context == nil {
+			context = map[string]any{}
+		}
+
+		// Don't overwrite this value if a user has somehow set it.
+		if _, ok := context[contextFlagKey]; !ok {
+			context[contextFlagKey] = flagKey
+		}
```

Comment:
I think we should do the opposite, and overwrite values in the `$flagd` property, but maybe warn when we do it.

Reply:
Good point. Updated. Thank you!